12_Unsupervised-learning.Rmd

```diff
@@ -0,0 +1,674 @@
+
+$$Z_m = \sum_{j=1}^p{\phi_{jm}X_j}=\phi_{1m}X_1+\phi_{2m}X_2+...+\phi_{pm}X_p$$
+
+And so, an approximation of the original features:
+
+$$x_{ij}\approx\sum_{m=1}^Mz_{im}\phi_{jm}$$
+
+
+We do this by **normalizing** these elements in the vector $z=\frac{x-\bar x}{sd(x)}$ to obtain a new vector of elements named **loadings**, $(\phi_{11},\phi_{21},...,\phi_{p1})^T$. In particular when $M=min(n-1,p)$, we have: $x_{ij}=\sum_{m=1}^Mz_{im}\phi_{jm}$.
+
+We *constrain* these **lodings** so their sum of squares would be equal to one:
+
+$$\sum_{j=1}^p{\phi_{j1}}^2=1$$
+
+**The Euclidean distance:**
+
+> "The first **M** principal component **score vectors** and the first **M** principal component **loading vectors** provide **the best M-dimensional approximation** (in terms of **Euclidean distance**) to the ith observation xij."
+
+
+$$\min_{}\left \{\sum_{j=1}^p\sum_{i=1}^n\left(x_{ij}-\sum_{m=1}^Mz_{im}\phi_{jm}\right)^2\right \}$$
+
+
+**The First pricipal component**
+
+Since we are only interested in **variance**, the total variance is defined as:
+
+$$\sum_{j=1}^pVar(X_j)=\sum_{j=1}^p\frac{1}{n}\sum_{i=1}^nx_{ij}^2$$
+
+The formula for the **max sample variance explained** by the *m*th principal component:
+
+$$\frac{1}{n}\sum_{i=1}^nz_{im}^2=\frac{1}{n}\sum_{i=1}^n\left(\sum_{j=1}^p\phi_{jm}x_{ij}\right)^2$$
+
+
+$$\max_{\phi_{11},\phi_{21},...,\phi_{p1}}\left \{\frac{1}{n}\sum_{i=1}^nz_{i1}^2\right \}$$
+$$\max_{\phi_{11},\phi_{21},...,\phi_{p1}}\left \{\frac{1}{n}\sum_{i=1}^n\left(\sum_{j=1}^p\phi_{j1}x_{ij}\right)^2\right \}$$
+$$subject \to\sum_{j=1}^p\phi_{j1}^2=1$$
+
+The calculation of this *objective* pass through the **eigen value decomposition** which alternative is the **singular value decomposition**.
+
+
+**How much of the information in a given data set is lost by projecting the observations onto the first few principal components?**
+
+**How much of the variance in the data is not contained in the first few principal components?**
+
+To answer this questions we need to consider the **proportion of variance explained (PVE)**
+
+Maximizing the variance of the first M principal components, we minimize the mean squared error of the M-dimensional approximation, and viceversa.
+
+In conclusion, principal component analysis is a question of **minimizing the approximation error or maximizing the variance**.
+
+
+## Geometric interpretation
+
+This is the representation of the Ad Spending vs Population with the first and second principal components.
+Let's go a bit more in deep about how to make such as visualization.
+```{r echo=FALSE, fig.dim="50%"}
+knitr::include_graphics("images/fig06-14.png")
+```
+
+source: http://www.sthda.com/english/articles/31-principal-component-methods-in-r-practical-guide/112-pca-principal-component-analysis-essentials/
+
+
+```{r message=FALSE, warning=FALSE, include=FALSE, paged.print=FALSE}
+library(tidymodels)
+library(tidyverse)
+library(magrittr)
+library(factoextra) # ?
```

Comment:
```suggestion
library(factoextra)
```

Why did this have a `?`? Perhaps related to this: Be sure to always run `usethis::use_package("factoextra")` (etc) in your console for any new package you use, so the system knows that that package is needed.

Reply:
ok